exec/java-exec/src/main/java/org/apache/drill/exec/work/user/UserWorker.java

```diff
@@ -73,9 +73,13 @@ private static QueryId queryIdGenerator() {
   }
 
   public QueryId submitWork(UserClientConnection connection, RunQuery query) {
+    return submitWork(connection, query, null);
+  }
+
+  public QueryId submitWork(UserClientConnection connection, RunQuery query, Integer autoLimitRowCount) {
```

Comment:
For the case when `RunQuery` will contain `autoLimitRowCount`, there will be no need for adding a new method.

Reply:
Agreed.. let me take a look at modifying RunQuery

Reply:
Done in the latest set of commits